miden-tx/src/tests/kernel_tests/test_tx.rs

```diff
@@ -820,57 +780,95 @@ fn test_load_foreign_account_basic() {
     );
 
     foreign_account_data_memory_assertions(&foreign_account, &process);
+
+    // GET ITEM TWICE
+    // --------------------------------------------------------------------------------------------
+    // Check the correctness of the memory layout after two consecutive invocations of the
+    // `get_item` account procedures. Invoking two foreign procedures from the same account should
+    // result in reuse of the loaded account.
+
+    let code = format!(
+        "
+        use.std::sys
+
+        use.kernel::prologue
+        use.miden::tx
+
+        begin
+            exec.prologue::prepare_transaction
+
+            {get_item_test_code} 
+            dropw
+            # => []
+
+            # call `get_item` again
+            {get_item_test_code}
+
+            # truncate the stack
+            exec.sys::truncate_stack
+        end
+        ",
+        get_item_test_code = get_item_test_code(&foreign_account_id),
+    );
+
+    let process = tx_context.execute_code(&code).unwrap();
+
+    assert_eq!(
+        try_read_root_mem_value(&process, NATIVE_ACCOUNT_DATA_PTR + ACCOUNT_DATA_LENGTH as u32 * 2),
+        None,
+        "Memory starting from 6144 should stay uninitialized"
+    );
 }
 
-/// This test checks that invoking two foreign procedures from the same account results in reuse of
-/// the loaded account.
 #[test]
-fn test_load_foreign_account_twice() {
+fn test_fpi_basic() {
```

Comment:
```suggestion
fn test_fpi_load_foreign_account_code() {
```

Nit: Suggestion for a more specific name for the test, but totally optional.

Reply:
I totally agree that the name of this test should be updated to be more specific, but I'm not sure about the name. In fact this is an integration test, which checks the correctness of the foreign procedure execution on every stage: we load the foreign account code, provide the mast forest to the executor, construct the account procedure maps and execute the foreign procedure to obtain the data from the foreign account's storage slot. So I think that `test_fpi_load_foreign_account_code` is not the best option, because it reflects only one of the execution stages. May be something like `test_fpi_execute_foreign_procedure`? So the logic is that the execution of the foreign procedure is the result achieved by the all previous stages, so it's kind of illustrating the accomplished work